packages/connectors/connector-dingtalk-web/src/index.ts

```diff
@@ -0,0 +1,162 @@
+  GetConnectorConfig,
+  CreateConnector,
+  SocialConnector,
+} from '@logto/connector-kit';
+import {
+  ConnectorError,
+  ConnectorErrorCodes,
+  validateConfig,
+  ConnectorType,
+  parseJson,
+} from '@logto/connector-kit';
+
+import {
+  authorizationEndpoint,
+  accessTokenEndpoint,
+  userInfoEndpoint,
+  scope as defaultScope,
+  defaultMetadata,
+  defaultTimeout,
+} from './constant.js';
+import type { DingtalkConfig } from './types.js';
+import {
+  dingtalkConfigGuard,
+  accessTokenResponseGuard,
+  userInfoResponseGuard,
+  authResponseGuard,
+} from './types.js';
+
+const getAuthorizationUri =
+  (getConfig: GetConnectorConfig): GetAuthorizationUri =>
+  async ({ state, redirectUri }) => {
+    const config = await getConfig(defaultMetadata.id);
+    validateConfig(config, dingtalkConfigGuard);
+
+    const { clientId, scope } = config;
+
+    const queryParameters = new URLSearchParams({
+      client_id: clientId,
+      redirect_uri: encodeURI(redirectUri), // The variable `redirectUri` should match {clientId, clientSecret}
+      response_type: 'code',
+      scope: scope ?? defaultScope,
+      state,
+      prompt: 'consent',
+    });
+
+    return `${authorizationEndpoint}?${queryParameters.toString()}`;
+  };
+
+export const getAccessToken = async (
+  code: string,
+  config: DingtalkConfig
+): Promise<{ accessToken: string }> => {
+  const { clientId, clientSecret } = config;
+
+  const httpResponse = await got.post(accessTokenEndpoint, {
+    json: {
+      clientId,
+      clientSecret,
+      code,
+      grantType: 'authorization_code',
+    },
+    timeout: { request: defaultTimeout },
+  });
+
+  const result = accessTokenResponseGuard.safeParse(parseJson(httpResponse.body));
+
+  if (!result.success) {
+    throw new ConnectorError(ConnectorErrorCodes.InvalidResponse, result.error);
+  }
+
+  const { accessToken } = result.data;
+  assert(accessToken, new ConnectorError(ConnectorErrorCodes.SocialAuthCodeInvalid));
+
+  return { accessToken };
+};
+
+const getUserInfo =
+  (getConfig: GetConnectorConfig): GetUserInfo =>
+  async (data) => {
+    const { code } = await authorizationCallbackHandler(data);
+    const config = await getConfig(defaultMetadata.id);
+    validateConfig(config, dingtalkConfigGuard);
+    const { accessToken } = await getAccessToken(code, config);
+
+    try {
+      const httpResponse = await got.get(userInfoEndpoint, {
+        headers: {
+          'x-acs-dingtalk-access-token': accessToken,
+        },
+        timeout: { request: defaultTimeout },
+      });
+      const rawData = parseJson(httpResponse.body);
+      const result = userInfoResponseGuard.safeParse(rawData);
+
+      if (!result.success) {
+        throw new ConnectorError(ConnectorErrorCodes.InvalidResponse, result.error);
+      }
+
+      const { nick: name, avatarUrl: avatar, unionId: id, email, mobile, stateCode } = result.data;
+      return {
+        id,
+        avatar,
+        phone: `${stateCode}${mobile}`,
```

Comment:
Since both `stateCode` and `mobile` are optional, I don't think we can simply append `stateCode` and `mobile`.